muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application/toolbar.tsx

```diff
@@ -144,6 +165,8 @@ class CommunicatorToolbar extends React.Component<CommunicatorToolbarProps, Comm
             }>
               <ButtonPill buttonModifiers="label" icon="tag"/>
             </Dropdown>
+            {isUnreadOrInboxOrLabel ? <ButtonPill buttonModifiers="toggle-read" icon={`message-${this.state.isCurrentRead ? "" : "un"}read`}
```

Comment:
This is ok for now but consider if icons name changes from icon-read/icon-unread to something completely different. Then we can't just prefix the icon suffix that just prefixes read to unread. Might be better to actually handle both cases separately so unread and read icon names are not related to each other.